source/Applications/Advanced/Downsample/Downsample.cpp

```diff
@@ -175,23 +31,24 @@ int main()
     {
         Zivid::Application zivid;
 
-        std::string fileName = std::string(ZIVID_SAMPLE_DATA_DIR) + "/Zivid3D.zdf";
-        std::cout << "Reading ZDF frame from file: " << fileName << std::endl;
-        const auto frame = Zivid::Frame(fileName);
+        std::string dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/Zivid3D.zdf";
+        std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+        const auto frame = Zivid::Frame(dataFile);
 
         const auto pointCloud = frame.pointCloud();
-        std::cout << "Loaded " << pointCloud.size() << " data points" << std::endl;
+        std::cout << "Size of point cloud before downsampling: " << pointCloud.size() << " data points" << std::endl;
+
+        std::cout << "Downsampling point cloud" << std::endl;
+        const auto downsampledPointCloud = pointCloud.downsampled(Zivid::PointCloud::Downsampling::by2x2);
```

Comment:
We also support in-place downsampling; `downsample()`. I think that is more relevant for the sample, since that is more performant and should be the default.

An good alternative is to show both in-place downsampling (`downsample()`) and downsampling into a new `Pointcloud` instance (`downsampled()`).

Reply:
Do you suggest moving the first visualizer call to before "downsample" then?

Reply:
That's how I solved it now. By moving the first viz

Reply:
> 
> 
> We also support in-place downsampling; `downsample()`. I think we should rather use that in this sample, since it is more performant (no copy) and should definitely be the default. @robertkt1
> 
> An alternative is to show both in-place downsampling (`downsample()`) and downsampling into a new `Pointcloud` instance (`downsampled()`).

@robertkt1 Rune's comment on two options should be covered in the point cloud tutorial: https://github.com/zivid/zivid-cpp-samples/blob/new-downsample-sample/source/Applications/PointCloudTutorial.md